### PR TITLE
Clean bazel mocks before generating them

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This PR fixes an issue where a stale Bazel cache could cause the `hack/copy-mocks.sh` file to error out.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by ashish-amarnath** [comment]
Do you mean `hack/copy-bazel-mocks.sh` will fail w/o this change?

File: Makefile [hunk 1]
```diff
@@ -266,6 +265,10 @@ clean: ## Remove all generated files
 clean-bazel: ## Remove all generated bazel symlinks
 	bazel clean
 
+.PHONY: clean-bazel-mocks
+clean-bazel-mocks: ## Remove all generated bazel mocks files
+	find bazel-bin/pkg -name '*_mock.go' -type f -delete
```
Comment (ashish-amarnath) [inline Makefile]: Can we use `bazel clean` here instead of `find -delete`?

Comment (vincepri) [reply Makefile]: The `make clean-bazel` target is too aggressive for this particular issue, this removes only the mocks in cache.

**Comment by vincepri** [comment]
@ashish-amarnath The target fails if your cache is stale, or even if you changed a folder name for example

File: Makefile (hunk 1 above)
Comment (ashish-amarnath) [reply Makefile]: Sorry for not being clear. I mean can we use `bazel clean <some-target>`? not `bazel clean` everything.

Comment (vincepri) [reply Makefile]: I don't think clean accepts any arguments from what I can see

Comment (ashish-amarnath) [reply Makefile]: OK. I wasn't able to find examples that we can use here.
I am 👍 w/ this change.

**Pull request merged**